src/library/scala/util/ChainingOps.scala

```diff
@@ -20,10 +22,7 @@ final class ChainingOps[A](val self: A) extends AnyVal {
    *  @tparam U     the result type of the function `f`.
    *  @return       the original value `self`.
    */
-  def tap[U](f: A => U): self.type = {
-    f(self)
-    self
-  }
+  def tap[U](f: A => U): A = macro ???
```

Comment:
I am not sure how to return `self.type` here using macro. If I just change the type to `self.type` I get:

```
util-chainingops.scala:12: error: type mismatch;
 found   : x$macro$1.type (with underlying type List[Int])
 required: _1.self.type where val _1: scala.util.ChainingOps[List[Int]]
        .tap(xs => x = xs.head)
            ^
one error found
```
 
When the macro is expanded `ChainingOps[List[Int]]` goes away, so.. can I still get `_1.self.type`?

Reply:
Yeah can't use self.type when using an enriched method.

You could do it if you did it as a synthetic method, like String.+